extension/chrome/settings/modules/contacts.ts

```diff
@@ -109,7 +109,7 @@ View.run(class ContactsView extends View {
   }
 
   private actionExportAllKeysHandler = () => {
-    const allArmoredPublicKeys = this.contacts.map(c => c.pubkey).filter(Boolean).map((c: Key) => (KeyUtil.armor(c)).trim()).join('\n');
+    const allArmoredPublicKeys = this.contacts.map(c => c.armoredPubkey).filter(Boolean).map(a => a!.trim()).join('\n');
```

Comment:
Depending on performance, I would consider to not even pull armored key out of the database when searching contacts this way. And then I want to view details about a contact like here, I would make another call to the database with `ContactStore.get(...)`.

That way, when I'm pulling many records from database - searching - I'm only just getting metadata, which should always be fast.

If it's already fast now, then that may be an unnecessary optimization, in which case we can leave it as is.

Reply:
I agree. I took a closer look and found out that pubkeys are only needed after the user presses "export all". But since our records don't have any ids, I introduced `ContactStore.searchPubkeys` method. What do you think?

Reply:
It's an improvement - at least there are two different methods now, which we can optimize easier later on.